app/models/miq_cockpit_ws_worker/runner.rb

```diff
@@ -0,0 +1,196 @@
+class MiqCockpitWsWorker::Runner < MiqWorker::Runner
+  BINDING_ADDRESS = ENV['BINDING_ADDRESS'] || (Rails.env.production? ? nil : "0.0.0.0")
+
+  # Copied from old MiqProcess method
+  LINUX_STATES = {
+    'I' => :idle,             # Marks a process that is idle (sleeping for longer than about 20 seconds).
+    'R' => :runnable,         # Marks a runnable process.
+    'S' => :sleeping,         # Marks a process that is sleeping for less than about 20 seconds.
+    'T' => :stopped,          # Marks a stopped process.
+    'D' => :waiting,          # Marks a process in uninterruptible wait.  (Linux)
+    'Z' => :zombie            # Marks a dead process (a ``zombie'').
+  }.freeze
+
+  # Copied from old MiqProcess method
+  def find_pids(cmd)
+    pids = []
+
+    Dir['/proc/[0-9]*/cmdline'].each do |filename|
+      cmdline = MiqSystem.readfile_async(filename)
+      next if cmdline.nil?
+      matches = cmd =~ cmdline.tr("\000", " ").strip
+      next if matches.nil?
+      pid = filename.split('/')[2]
+      pids << pid.to_i
```

Comment:
All of this stuff should be done with Sys::ProcTable, similar to this:  https://github.com/ManageIQ/manageiq-gems-pending/blob/732f36463f965672ce0de0522b31c8fe5aaea9c5/lib/gems/pending/util/miq-process.rb#L129

Something like this:

```ruby

pids = Set.new
Sys::ProcTable.ps.each |process_stuct|
  pids << process_struct.pid if cmd =~ (process_struct.try(:cmdline) || "")
end
```